apps/supaclk/app.js

```diff
@@ -0,0 +1,195 @@
+    32,
+    atob("BAUHCwsNEQQGBgkKBQkFBwwHCgkKCQoJCgoFBQsLCwkQDAwNDgwLDQ8HBw0LEQ0PDA8NCwwMDBINCwsGBwY="),
+    18|65536
+  );
+}
+
+// Actually draw the watch face
+let draw = function() {
+  g.reset().setColor(g.theme.fg).setBgColor(g.theme.bg);
+  g.clearRect(R.x,R.y, R.x2, R.y2 - 60);
+  var date = new Date();
+  let minutes = date.getMinutes();
+  minutes = (minutes < 10 ? '0' : '') + minutes;
+
+  g.setColor(g.theme.fg)
+  // Time
+  const yt =  R.y + 92 - 20 - 30 + 6 + 10;
+  const xt = R.w/2 - 5;
+  let hours = date.getHours()+'';
+  g.setFontAlign(1, 0).setFontPlayfairDisplay().drawString(hours, xt - 8, yt);
+  g.setFontAlign(0, 0).setFontPlayfairDisplay().drawString(':', xt, yt);
+  g.setFontAlign(-1, 0).setFontPlayfairDisplay().drawString(minutes, xt + 8, yt);
+  // logo
+  g.drawImage(supaClockImg, R.x2 - supaClockImg.width - 2, R.y + 2);
+  // dow + date
+  let dateStr = require("locale").dow(date).toUpperCase() + '\n' + require("locale").month(date, 2).toUpperCase() + ' ' + date.getDate();
+  g.setFont('6x8').setFontAlign(-1, 0).drawString(dateStr, R.x2 - supaClockImg.width - 2, R.y + 42 - 30 + 8);
+
+  if (drawTimeout) clearTimeout(drawTimeout);
+  drawTimeout = setTimeout(function() {
+    drawTimeout = undefined;
+    draw();
+  }, 60000 - (Date.now() % 60000));
+};
+
+let clockInfoDraw = (itm, info, options) => {
+  g.reset().setFontPlayfairDisplaySm().setColor(g.theme.fg).setBgColor(g.theme.bg);
+  if (options.focus) g.setBgColor("#FF0");
+  g.clearRect({x:options.x,y:options.y,w:options.w,h:options.h});
+
+  if (info.img) {
+    g.drawImage(info.img, options.x+1,options.y+2);
+  }
+  var text = info.text.toString().toUpperCase();
+  if (g.setFontPlayfairDisplaySm().stringWidth(text)+24-2>options.w) g.setFont("8x12");
```

Comment:
I get there errors, using firmware 2v22.5 (have not tried stable 2v22 for now):

```
Uncaught Error: Unknown font "8x12"
 at line 36 col 675 in supaclk.app.js
...>options.w)g.setFont("8x12");g.setFontAlign(-1,-1).drawStrin...
                              ^
in function "draw" called from line 11 col 3260 in clock_info
options.draw(itm,itm.get(),options);
                                  ^
in function "drawItem" called from line 11 col 3319 in clock_info
drawItem(itm)
            ^
in function called from system
 at line 36 col 675 in supaclk.app.js
...>options.w)g.setFont("8x12");g.setFontAlign(-1,-1).drawStrin...
                              ^
in function "draw" called from line 11 col 3260 in clock_info
options.draw(itm,itm.get(),options);
                                  ^
in function "drawItem" called from line 11 col 3319 in clock_info
drawItem(itm)
            ^
in function called from system
 at line 36 col 1155 in supaclk.app.js
...>options.w)g.setFont("8x12");g.setFontAlign(1,-1).drawString...
                              ^
in function "draw" called from line 11 col 3260 in clock_info
options.draw(itm,itm.get(),options);
                                  ^
in function "drawItem" called from line 11 col 3319 in clock_info
drawItem(itm)
            ^
in function called from system
```

Looking at [the `g.setFont` reference](https://www.espruino.com/ReferenceBANGLEJS2#l_Graphics_setFont) I don't think `"8x12"` is available. But you could do `g.setFont("Vector:12")` or `g.setFont("4x6:2")` to maybe achieve what you want?